django/db/models/fields/__init__.py

```diff
@@ -1085,6 +1085,8 @@ def formfield(self, **kwargs):
         # will be validated twice. This is considered acceptable since we want
         # the value in the form field (to pass into widget for example).
         defaults = {'max_length': self.max_length}
+        if self.null and not connection.features.interprets_empty_strings_as_nulls:
```

Comment:
For example, we could try to figure out the database engine to use by asking the router about `db_for_read(self.model)` and `db_for_write(self.model)` but what should be done when both engines differ?